tests/dataset_generator_test.py

```diff
@@ -946,3 +948,25 @@ def test_dataset_generator_terminates(mocked_generate_example):
     generated_df = generated_dataset.to_pandas()
     assert len(generated_dataset) == 100
     assert list(generated_df.columns) == ["input_col", "output_col"]
+
+
+def test_generate_dataset_agent_switch():
+    """Test if test set generation can switch between API agents."""
```

Comment:
What do you mean by "switch between API agents"? Current context implies that "we can use different Agents to generate datasets", but not "generate some examples from Agent A, the switch to Agent B to generate more".

Reply:
I personally think that the words "switch between" is a bit unclear.